openedx/core/djangoapps/cache_toolbox/tests/test_middleware.py

```diff
@@ -37,8 +40,9 @@ def _test_change_session_hash(self, test_url, redirect_url, target_status_code=2
         response = self.client.get(test_url)
         assert response.status_code == 200
         with patch.object(User, 'get_session_auth_hash', return_value='abc123'):
-            response = self.client.get(test_url)
-            self.assertRedirects(response, redirect_url, target_status_code=target_status_code)
+            with patch.object(User, '_legacy_get_session_auth_hash', return_value='abc123'):
+                response = self.client.get(test_url)
+                self.assertRedirects(response, redirect_url, target_status_code=target_status_code)
```

Comment:
Is this patching the only difference between the Django version tests? Could something like the following get rid of most of the test duplication?
```suggestion
            # Django 3.2 has _legacy_get_session_auth_hash, and Django 4 does not
            if hasattr(User, '_legacy_get_session_auth_hash'):  # not sure if this works, or something like this????
                with patch.object(User, '_legacy_get_session_auth_hash', return_value='abc123'):
                    response = self.client.get(test_url)
            else:
                response = self.client.get(test_url)

            self.assertRedirects(response, redirect_url, target_status_code=target_status_code)
```

Also, maybe adding `autospec=True` to the first `patch.object` will also help protect you more?

Reply:
Updated